libs/threadmanager/src/threadmanager.cpp

```diff
@@ -363,21 +367,25 @@ namespace hpx { namespace threads {
                 // set parameters for scheduler and pool instantiation and
                 // perform compatibility checks
                 hpx::detail::ensure_high_priority_compatibility(cfg_.vm_);
-                std::string affinity_desc;
-                std::size_t numa_sensitive =
-                    hpx::util::get_affinity_description(cfg_, affinity_desc);
 
                 // instantiate the scheduler
                 using local_sched_type =
                     hpx::threads::policies::local_queue_scheduler<>;
 
                 local_sched_type::init_parameter_type init(
                     thread_pool_init.num_threads_,
-                    thread_pool_init.affinity_data_, numa_sensitive,
+                    thread_pool_init.affinity_data_,
                     thread_queue_init, "core-local_queue_scheduler");
+
                 std::unique_ptr<local_sched_type> sched(
                     new local_sched_type(init));
 
+                // set the default scheduler flags
+                sched->add_scheduler_mode(thread_pool_init.mode_);
+                // conditionally set/unset this flag
+                sched->update_scheduler_mode(
+                    policies::enable_stealing_numa, !numa_sensitive);
+
```

Comment:
Could we pass the default mode directly to the constructor?

Reply:
Yes. The PR is still slightly WIP and there are a few missing things that I'm cleaning up and fixing now. The default mode was missing in the last commit, but fixed in the next one.
The numa sensitive mode is not default because the command line can override it, so by default be allow numa-stealing, but it can be turned off, the update_mode(blah, false) does this.